test/sample-test/run_test.sh

```diff
@@ -101,242 +103,217 @@ while [ "$1" != "" ]; do
+  sed -i "s|gcr.io/ml-pipeline/ml-pipeline-local-roc:\([a-zA-Z0-9_.-]\)\+|${LOCAL_ROC_IMAGE}|g" ${TEST_NAME}.yaml
+}
 
-  # Compile samples
-  cd ${BASE_DIR}/samples/core/recursion
-  dsl-compile --py recursion.py --output recursion.tar.gz
+if [[ -z "$RESULTS_GCS_DIR" ]]; then
+  usage
+  exit 1
+fi
 
-  cd "${TEST_DIR}"
-  python3 run_basic_test.py --input ${BASE_DIR}/samples/core/recursion/recursion.tar.gz --result $SAMPLE_RECURSION_TEST_RESULT --output $SAMPLE_RECURSION_TEST_OUTPUT --testname recursion --namespace ${NAMESPACE}
+if [[ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]]; then
+  gcloud auth activate-service-account \
+  --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+fi
 
-  echo "Copy the test results to GCS ${RESULTS_GCS_DIR}/"
-  gsutil cp ${SAMPLE_RECURSION_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_RECURSION_TEST_RESULT}
-elif [ "$TEST_NAME" == "xgboost" ]; then
-  SAMPLE_XGBOOST_TEST_RESULT=junit_SampleXGBoostOutput.xml
-  SAMPLE_XGBOOST_TEST_OUTPUT=${RESULTS_GCS_DIR}
+cd ${BASE_DIR}
 
-  # Compile samples
-  cd ${BASE_DIR}/samples/core/xgboost-spark
+# Install argo
+echo "install argo"
+ARGO_VERSION=v2.3.0
+mkdir -p ~/bin/
+export PATH=~/bin/:$PATH
+curl -sSL -o ~/bin/argo "https://github.com/argoproj/argo/releases/download/$ARGO_VERSION/argo-linux-amd64"
+chmod +x ~/bin/argo
 
-  dsl-compile --py xgboost-training-cm.py --output xgboost-training-cm.yaml
+echo "Run the sample tests..."
 
-  if [ -n "${DATAPROC_CREATE_CLUSTER_IMAGE}" ];then
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataproc-create-cluster:\([a-zA-Z0-9_.-]\)\+|${DATAPROC_CREATE_CLUSTER_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataproc-delete-cluster:\([a-zA-Z0-9_.-]\)\+|${DATAPROC_DELETE_CLUSTER_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataproc-analyze:\([a-zA-Z0-9_.-]\)\+|${DATAPROC_ANALYZE_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataproc-transform:\([a-zA-Z0-9_.-]\)\+|${DATAPROC_TRANSFORM_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataproc-train:\([a-zA-Z0-9_.-]\)\+|${DATAPROC_TRAIN_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataproc-predict:\([a-zA-Z0-9_.-]\)\+|${DATAPROC_PREDICT_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-roc:\([a-zA-Z0-9_.-]\)\+|${LOCAL_ROC_IMAGE}|g" xgboost-training-cm.yaml
+# Run the tests
+preparation ${TEST_NAME}
+
+if [[ "${TEST_NAME}" == "kubeflow_training_classification" ]]; then
+  #TODO(numerology): convert the sed commands to sed -e
+  # 's|gcr.io/ml-pipeline/|gcr.io/ml-pipeline-test/' and tag replacement. Also
+  # let the postsubmit tests refer to yaml files.
+
```

Comment:
Could we keep the sed command for the kubeflow_training_classification as well?

Reply:
https://github.com/kubeflow/pipelines/blob/1407d86497bc0de3e4a20d5a6bb9689dc6085346/test/sample-test/run_test.sh#L143

Reply:
Agreed. Done.